web/analytics/views.py

```
@@ -0,0 +1,427 @@
+    else:
+        return HttpResponse("Unsupported export format", status=400)
+
+@login_required
+@user_passes_test(is_educator)
+def educator_analytics_dashboard(request):
+    """
+    Main analytics dashboard for educators showing comprehensive data and insights.
+    """
+    # Get the teacher's courses
+    courses = Course.objects.filter(teacher=request.user)
+    course_ids = courses.values_list('id', flat=True)
```

Ruff (0.8.2)

Local variable `course_ids` is assigned to but never used

Remove assignment to unused variable `course_ids`

(F841)